src/NextGen/DonationForm/Blocks/DonationFormBlock/app/templates/fields/Amount.tsx

```diff
@@ -0,0 +1,39 @@
+import {FieldProps} from '../index';
+import {useMemo} from 'react';
+// import {useFormContext} from 'react-hook-form';
+
+interface AmountProps extends FieldProps {
+    levels: bigint[];
+    allowCustomAmount: boolean;
+}
+
+export default function Amount({name, label, inputProps, levels, allowCustomAmount}: AmountProps) {
+    const {useFormContext} = window.givewp.form;
+    const {setValue, watch} = useFormContext();
+    const currency = watch('currency');
```

Comment:
Very strange. I've switched to `useWatch`, but I'm still seeing the whole form re-render when I click a button. I doubt that's `useWatch`, though, as the currency never changes. Somehow `setValue` is triggering the whole re-render. Hmm.

I'm not seeing the form submission, though. That's new to me. Perhaps let's hop on a call and see if we can figure this out together, when you have time.

Reply:
@JasonTheAdams I was able to fix a few things:

1. [7d2a3f7](https://github.com/impress-org/givewp-next-gen/pull/18/commits/7d2a3f7215a255540b76b137eefaad7f30987f6f) Every button inside the form needs a `type=button` attribute or else it will override the main submit button
2. [277f58e](https://github.com/impress-org/givewp-next-gen/pull/18/commits/277f58ec8ec71ba20c2a86fef8972213ec1f1b08) We need to prevent as much potential for re-rendering inside the form component as possible.  Moving the FormTemplate render component outside of the Form component is necessary to preserve form state on submit and reduce re-renders.  Without doing this if you get an error using Stripe it would jump back to selecting the original default gateway (test gateway)